src/main/java/internet/shop/service/impl/ProductServiceImpl.java

```diff
@@ -0,0 +1,46 @@
+package internet.shop.service.impl;
+
+import internet.shop.dao.ProductDao;
+import internet.shop.lib.Inject;
+import internet.shop.lib.Service;
+import internet.shop.model.Product;
+import internet.shop.service.ProductService;
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class ProductServiceImpl implements ProductService {
+    @Inject
+    private ProductDao productDao;
+
+    @Override
+    public Product createProduct(String name, Double price) {
+        return productDao.createProduct(name, price);
+    }
+
+    @Override
+    public void addProduct(Product product) {
+        productDao.addProduct(product);
+    }
+
+    @Override
+    public Optional<Product> getProduct(Long id) {
```

Comment:
In services, get methods should return Product, not Optional<Product>

Reply:
Fixed :) Thank you